coralproject/talk
# [CORL-2660] Mitigate duplicate story ID errors by updating story url if a valid id is provided

## What does this PR do?

During the retrieval of a comment stream via the createStreamEmbed, if the site provides a valid storyID that matches a real story but the url it is coming from differs from the one on record, we update the story url.

This prevents numerous "DUPLICATE_STORY_ID" errors when newsrooms modify their story URL multiple times in their CMS due to article title changes, slug changes, etc.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags? 

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- create a story in `multi-site-test`
- navigate to story and see that it creates the new story
- post a unique comment you'll remember
- using the id of the story you just created, navigate to `/story/alt/<STORY_ID>` instead
- see that it loads the same comment stream you just created despite the URL being different and has the comment you made previously
- check Mongo and query that story id, see that its URL is now the new `/alt/` URL
- go back to the old URL without the `/alt/` part
- see that is again still loads the stream with your comment on it
- check Mongo and query that story id, see that its URL is now back to the old URL without the `/alt/`
 
## How do we deploy this PR?

No special considerations
